numpy/f2py/tests/test_crackfortran.py

```diff
@@ -86,3 +86,25 @@ def test_defaultPublic(self, tmp_path):
         assert 'public' not in mod['vars']['a']['attrspec']
         assert 'private' not in mod['vars']['seta']['attrspec']
         assert 'public' in mod['vars']['seta']['attrspec']
+
+
+class TestArrayDimCalculation(util.F2PyTest):
+    # Issue gh-8062.  Calculations that occur in the dimensions of fortran
+    # array declarations should be interpreted by f2py as integers not floats.
+    # Prior to fix, test fails as generated fortran wrapper does not compile.
+    code = """
+        function test(n, a)
+          integer, intent(in) :: n
+          real(8), intent(out) :: a(0:2*n/2)
```

Comment:
What shape should `a(0:n/2*2)` give when n is odd?

Reply:
Good question. Fortran uses integer maths is array declarations, so that shape here, in Python syntax, is `(1+2*(n//2),)`

I have submitted a new PR (gh-17687) to fix the original problem and it includes a test of this example.

Reply:
What does `2*(-n)/(-2)` give in fortran?